"##[error]Cannot bind argument to parameter 'Path' because it is null." on self-hosted agent

With this in the pipeline:
```yaml
steps:
- task: DavidGardiner.chocolatey-tool-installer-task.chocolatey-tool-installer-task.ChocolateyToolInstaller@1
  inputs:
    force: false
```

I get the following debug output:
```
2020-09-14T16:06:51.6927983Z ##[debug]Evaluating condition for step: 'ChocolateyToolInstaller'
2020-09-14T16:06:51.6929035Z ##[debug]Evaluating: SucceededNode()
2020-09-14T16:06:51.6929265Z ##[debug]Evaluating SucceededNode:
2020-09-14T16:06:51.6929619Z ##[debug]=> True
2020-09-14T16:06:51.6929875Z ##[debug]Result: True
2020-09-14T16:06:51.6930185Z ##[section]Starting: ChocolateyToolInstaller
2020-09-14T16:06:51.7018121Z ==============================================================================
2020-09-14T16:06:51.7018352Z Task         : Chocolatey tool installer
2020-09-14T16:06:51.7018648Z Description  : Acquires the latest version of Chocolatey from the internet and adds it to the PATH (if not already installed). This task only works on Windows agents
2020-09-14T16:06:51.7018919Z Version      : 1.0.6
2020-09-14T16:06:51.7019056Z Author       : David Gardiner
2020-09-14T16:06:51.7019468Z Help         : Runs the standard [Chocolatey installer](https://chocolatey.org/install) to install Chocolatey on the build agent.
Microsoft-hosted agents come with Chocolatey pre-installed, but self-hosted agents may not so this task ensures Chocolatey is always available for use.
2020-09-14T16:06:51.7019945Z ==============================================================================
2020-09-14T16:06:52.2641100Z ##[debug]VstsTaskSdk 0.11.0 commit 7ff27a3e0bdd6f7b06690ae5f5b63cb84d0f23f4
2020-09-14T16:06:52.3330292Z ##[debug]Entering C:\AzureDevopsAgent\_work\_tasks\ChocolateyToolInstaller_1dc7ea30-2588-4837-9c45-f731e537d175\1.0.6\installer.ps1.
2020-09-14T16:06:52.3458386Z ##[debug]Caught exception from task script.
2020-09-14T16:06:52.3478418Z ##[debug]Error record:
2020-09-14T16:06:52.3936123Z ##[debug]ForEach-Object : Cannot bind argument to parameter 'Path' because it is null.
2020-09-14T16:06:52.3944838Z ##[debug]At line:19 char:13
2020-09-14T16:06:52.3953585Z ##[debug]+             ForEach-Object {
2020-09-14T16:06:52.3962179Z ##[debug]+             ~~~~~~~~~~~~~~~~
2020-09-14T16:06:52.3970965Z ##[debug]    + CategoryInfo          : InvalidData: (:) [ForEach-Object], ParameterBindingValidationException
2020-09-14T16:06:52.3979378Z ##[debug]    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.ForEachObje    ctCommand
2020-09-14T16:06:52.3987715Z ##[debug] 
2020-09-14T16:06:52.4002383Z ##[debug]Script stack trace:
2020-09-14T16:06:52.4027868Z ##[debug]at <ScriptBlock>, C:\AzureDevopsAgent\_work\_tasks\ChocolateyToolInstaller_1dc7ea30-2588-4837-9c45-f731e537d175\1.0.6\installer.ps1: line 6
2020-09-14T16:06:52.4036140Z ##[debug]at <ScriptBlock>, <No file>: line 1
2020-09-14T16:06:52.4044396Z ##[debug]at <ScriptBlock>, <No file>: line 22
2020-09-14T16:06:52.4052916Z ##[debug]at <ScriptBlock>, <No file>: line 18
2020-09-14T16:06:52.4061393Z ##[debug]at <ScriptBlock>, <No file>: line 1
2020-09-14T16:06:52.4075023Z ##[debug]Exception:
2020-09-14T16:06:52.4102780Z ##[debug]System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'Path' because it is null.
2020-09-14T16:06:52.4110452Z ##[debug]   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
2020-09-14T16:06:52.4118591Z ##[debug]   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
2020-09-14T16:06:52.4126926Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2020-09-14T16:06:52.4135171Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2020-09-14T16:06:52.4143326Z ##[debug]   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
2020-09-14T16:06:52.4151562Z ##[debug]   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
2020-09-14T16:06:52.4160107Z ##[debug]   at System.Management.Automation.ScriptBlock.InvokeWithPipeImpl(ScriptBlockClauseToInvoke clauseToInvoke, Boolean createLocalScope, Dictionary`2 functionsToDefine, List`1 variablesToDefine, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo invocationInfo, Object[] args)
2020-09-14T16:06:52.4168154Z ##[debug]   at System.Management.Automation.ScriptBlock.<>c__DisplayClass57_0.<InvokeWithPipe>b__0()
2020-09-14T16:06:52.4176307Z ##[debug]   at System.Management.Automation.Runspaces.RunspaceBase.RunActionIfNoRunningPipelinesWithThreadCheck(Action action)
2020-09-14T16:06:52.4184814Z ##[debug]   at System.Management.Automation.ScriptBlock.InvokeWithPipe(Boolean useLocalScope, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo invocationInfo, Boolean propagateAllExceptionsToTop, List`1 variablesToDefine, Dictionary`2 functionsToDefine, Object[] args)
2020-09-14T16:06:52.4192946Z ##[debug]   at System.Management.Automation.ScriptBlock.InvokeUsingCmdlet(Cmdlet contextCmdlet, Boolean useLocalScope, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Object[] args)
2020-09-14T16:06:52.4200981Z ##[debug]   at Microsoft.PowerShell.Commands.ForEachObjectCommand.ProcessRecord()
2020-09-14T16:06:52.4209459Z ##[debug]   at System.Management.Automation.CommandProcessor.ProcessRecord()
2020-09-14T16:06:52.4378753Z ##[error]Cannot bind argument to parameter 'Path' because it is null.
2020-09-14T16:06:52.4384812Z ##[debug]Processed: ##vso[task.logissue type=error]Cannot bind argument to parameter 'Path' because it is null.
2020-09-14T16:06:52.4388947Z ##[debug]Processed: ##vso[task.complete result=Failed]
2020-09-14T16:06:52.4598254Z ##[section]Finishing: ChocolateyToolInstaller
```

--- a/README.md
+++ b/README.md
@@ -2,10 +2,10 @@
 
 This is an Azure DevOps task for Azure Pipelines to install [Chocolatey](https://chocolatey.org).
 
-Whilst Microsoft-hosted agents come with Chocolatey preinstalled, self-hosted agents may not. Including this task ensures Chocolatey
+Whilst Microsoft-hosted agents come with Chocolatey pre-installed, self-hosted agents may not. Including this task ensures Chocolatey
 is always installed regardless of the agent type.
 
 It was inspired in part by the [Chocolatey extension for Azure DevOps](https://github.com/chocolatey-community/chocolatey-azuredevops),
-and is a complimentary in that the one thing that task doesn't handle is installation.
+and in fact this task has since been incorporated into that extension.
 
-See [Overview](overview.md) for installation and useage instructions.
+See [Overview](overview.md) for installation and usage instructions.

--- a/installerTask/installer.ps1
+++ b/installerTask/installer.ps1
@@ -3,10 +3,12 @@
 Trace-VstsEnteringInvocation $MyInvocation
 
 $chocoInstallLocation = [Environment]::GetEnvironmentVariable("ChocolateyInstall", "Machine")
-if(-not (Test-Path $chocoInstallLocation)) {
+if ((-not $chocoInstallLocation) -or ($chocoInstallLocation -and -not (Test-Path $chocoInstallLocation))) {
     Write-Output "Environment variable 'ChocolateyInstall' was not found in the system variables. Attempting to find it in the user variables..."
     $chocoInstallLocation = [Environment]::GetEnvironmentVariable("ChocolateyInstall", "User")
 }
+
+Write-Verbose "chocoInstallLocation: $chocoInstallLocation"
 
 $chocoExe = "$chocoInstallLocation\choco.exe"
 

--- a/vss-extension.json
+++ b/vss-extension.json
@@ -2,7 +2,7 @@
     "manifestVersion": 1,
     "id": "chocolatey-tool-installer-task",
     "name": "Chocolatey Installer",
-    "version": "1.0.6",
+    "version": "1.1.0",
     "publisher": "DavidGardiner",
     "public": true,
     "targets": [

